src/compression/index.js

```diff
@@ -1,29 +1,24 @@
-import RawDecoder from './raw';
-import LZWDecoder from './lzw';
-import JpegDecoder from './jpeg';
-import DeflateDecoder from './deflate';
-import PackbitsDecoder from './packbits';
-import LercDecoder from './lerc';
+const registry = new Map();
 
-export function getDecoder(fileDirectory) {
-  switch (fileDirectory.Compression) {
-    case undefined:
-    case 1: // no compression
-      return new RawDecoder();
-    case 5: // LZW
-      return new LZWDecoder();
-    case 6: // JPEG
-      throw new Error('old style JPEG compression is not supported.');
-    case 7: // JPEG
-      return new JpegDecoder(fileDirectory);
-    case 8: // Deflate as recognized by Adobe
-    case 32946: // Deflate GDAL default
-      return new DeflateDecoder();
-    case 32773: // packbits
-      return new PackbitsDecoder();
-    case 34887: // LERC
-      return new LercDecoder(fileDirectory);
-    default:
-      throw new Error(`Unknown compression method identifier: ${fileDirectory.Compression}`);
+export function addDecoder(cases, importFn) {
+  if (!Array.isArray(cases)) cases = [cases];
+  cases.forEach(c => registry.set(c, importFn));
+}
+
+export async function getDecoder(fileDirectory) {
+  const importFn = registry.get(fileDirectory.Compression);
+  if (!importFn) {
+    throw new Error(`Unknown compression method identifier: ${fileDirectory.Compression}`);
   }
+  const Decoder = await importFn();
+  return new Decoder(fileDirectory);
```

Comment:
How efficient is the `import` here? I was wondering whether we should cache the importing to improve performance?

Reply:
Hi sorry for the delay in response here. 

In the browser, this would rely on the browser caching of JS modules. I suppose you could add a top level cache in geotiff, but ultimately a user could implement this if they were focused on performance (overriding the default that uses a bare dynamic import). 

In addition, dynamic imports are statically analyzable by via bundlers. Rollup has the `--inlineDynamicImports` flag (which would change each of the bare `import()`), and you can also customize how each dynamic import is handled: https://rollupjs.org/guide/en/#resolvedynamicimport